elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLColumn.java

```diff
@@ -5,18 +5,131 @@
+     */
+    List<JoinPath> getJoinPaths();
+
+    /**
+     * SQL Column would resolve references in {@link DimensionFormula} when constructing physical reference.
+     *
+     * @return {@link LabelResolver} for sql columns
+     */
+    default LabelResolver constructSQLColumnLabelResolver(EntityDictionary dictionary) {
+        Class<?> tableClass = dictionary.getEntityClass(getColumn().getTable().getId());
+        String fieldName = getColumn().getName();
+
+        DimensionFormula formula = dictionary.getAttributeOrRelationAnnotation(
+                tableClass, DimensionFormula.class, fieldName);
+
+        if (formula == null) {
+            JoinTo joinTo = dictionary.getAttributeOrRelationAnnotation(tableClass, JoinTo.class, fieldName);
+
+            return joinTo == null || joinTo.path().equals("")
+                    ? getLogicalColumnResolver(tableClass, fieldName)
+                    : getJoinToResolver(tableClass, joinTo);
+        } else {
+            return getFormulaResolver(tableClass, formula);
+        }
+    }
+
+    /**
+     * Get a logical column reference resolver
+     *
+     * @param tableClass table class
+     * @param fieldName field name
+     * @return a resolver
+     */
+    default LabelResolver getLogicalColumnResolver(Class<?> tableClass, String fieldName) {
+        return new LabelResolver(getColumn()) {
+            @Override
+            public String resolveLabel(LabelStore labelStore, String tableAlias) {
+                return getFieldAlias(
+                        tableAlias,
+                        labelStore.getDictionary().getAnnotatedColumnName(tableClass, fieldName));
+            }
+        };
+    }
+
+    /**
+     * Get a {@link JoinTo} reference resolver.
+     *
+     * @param tableClass table class
+     * @param joinTo join to path
+     * @return a resolver
+     */
+    default LabelResolver getJoinToResolver(Class<?> tableClass, JoinTo joinTo) {
+        return new LabelResolver(getColumn()) {
+            @Override
+            public Set<JoinPath> resolveJoinPaths(LabelStore labelStore, JoinPath from) {
+                JoinPath to = getJoinToPath(labelStore);
+
+                return labelStore.getLabelResolver(to).resolveJoinPaths(labelStore, extendJoinPath(from, to));
+            }
+
+            @Override
+            public Set<LabelResolver> getDependencyResolvers(LabelStore labelStore) {
+                return Collections.singleton(labelStore.getLabelResolver(getJoinToPath(labelStore)));
+            }
+
+            @Override
+            public String resolveLabel(LabelStore labelStore, String tableAlias) {
+                JoinPath joinToPath = getJoinToPath(labelStore);
+
+                return labelStore.resolveLabel(joinToPath, extendTypeAlias(tableAlias, joinToPath));
+            }
+
+            private JoinPath getJoinToPath(LabelStore labelStore) {
+                return new JoinPath(tableClass, labelStore.getDictionary(), joinTo.path());
```

Comment:
The code is harder to understand when we wrap constructors in one line functions.   I would remove these.